### PR TITLE
Export validation separately

### PR DESCRIPTION
**Pull request by martonmoro**

### CODE REVIEW

File: examples/unique-hash.eg.ts
Comment (mitschabaude) [inline examples/unique-hash.eg.ts:1]: can you revert changes to this example?

File: src/validation/index.ts
Comment (mitschabaude) [inline src/validation/index.ts:1]: is this file necessary? if not, can you remove it and export from `validation.ts`?

File: src/validation/validation.ts
Comment (mitschabaude) [inline src/validation/validation.ts:1]: was it necessary to move this file? if not, can you move it back?

File: src/validation/validation.ts [hunk 1]
```diff
@@ -1,8 +1,5 @@
 import { record, z } from 'zod';
-import type {
-  SerializedNestedType,
-  SerializedType,
-} from './serialize-provable.ts';
```
Comment (mitschabaude) [inline src/validation/validation.ts:5]: now we have these types twice, can you create a third file `serialize-provable-types.ts` and import them from there in both of the other files?

File: src/validation/validation.ts [hunk 2]
```diff
@@ -124,8 +164,28 @@ const SerializedSignatureSchema = z
   .strict();
 
 // Node schemas
-
-const NodeSchema: z.ZodType<any> = z.lazy(() =>
+type Node =
```
Comment (mitschabaude) [inline src/validation/validation.ts:167]: deduplicate this type

**Review by mitschabaude — COMMENTED** [review]
some comments!

File: src/validation/validation.ts (hunk 1 above)
Comment (mitschabaude) [reply src/validation/validation.ts:5]: (assuming that importing them from the original file causes an unwanted o1js dependency)

File: src/validation/validation.ts (hunk 2 above)
Comment (martonmoro) [reply src/validation/validation.ts]: This `Node` type is for the serialized Nodes. It is different from the `Node` type in operation.ts. Did you mean that?

Comment (mitschabaude) [reply src/validation/validation.ts]: ah yes I meant that. ignore the comment in that case!

**Review by mitschabaude — APPROVED** [review]
approving, will publish from this PR
